notebooks/ch12/linear_projection.md

```diff
@@ -0,0 +1,330 @@
+
+
+When $\hat{\theta_0} = 2$ and $\hat{\theta_1} = 0.5$, we add half of $x$ with two times the unit vector, resulting in a new vector.
+
+
+```python
+# HIDDEN
+plt.quiver([0, 0, 0], [0, 0, 0], [1, 2, 3], [1, -4, 0], 
+           angles='xy', scale_units='xy', scale=1, 
+           color=['black', 'black', 'red'])
+plt.xlim(-7, 7)
+plt.ylim(-8, 8)
+plt.show()
+```
+
+
+![png](linear_projection_files/linear_projection_14_0.png)
+
+
+When $\hat{\theta_0} = 3$ and $\hat{\theta_1} = -1$, we add $3$ times the unit vector to $-1$ times $x$, resulting in yet another vector.
+
+
+```python
+# HIDDEN
+plt.quiver([0, 0, 0, 0], [0, 0, 0, 0], [1, 2, 3, 1], [1, -4, 0, 7], 
+           angles='xy', scale_units='xy', scale=1, 
+           color=['black', 'black', 'blue', 'red'])
+plt.xlim(-7, 7)
+plt.ylim(-8, 8)
+plt.show()
+```
+
+
+![png](linear_projection_files/linear_projection_16_0.png)
+
+
+With $\hat{\theta_0} = -2$ and $\hat{\theta_1} = -1$, we get:
+
+
+```python
+# HIDDEN
+plt.quiver([0, 0, 0, 0, 0], [0, 0, 0, 0, 0], [1, 2, 3, 1, -4], [1, -4, 0, 7, 2], 
+           angles='xy', scale_units='xy', scale=1, 
+           color=['black', 'black', 'blue', 'blue', 'red'])
+plt.xlim(-7, 7)
+plt.ylim(-8, 8)
+plt.show()
+```
+
+
+![png](linear_projection_files/linear_projection_18_0.png)
+
+
+With $\hat{\theta_0} = -4$ and $\hat{\theta_1} = 1$, we get:
+
+
+```python
+# HIDDEN
+plt.quiver([0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0], [1, 2, 3, 1, -4, -2], [1, -4, 0, 7, 2, -8], 
+           angles='xy', scale_units='xy', scale=1, 
+           color=['black', 'black', 'blue', 'blue', 'blue', 'red'])
+plt.xlim(-7, 7)
+plt.ylim(-8, 8)
+plt.show()
+```
+
+
+![png](linear_projection_files/linear_projection_20_0.png)
+
+
+This process can go on and on, showing that when we can use any scalar multiples, we can make an infinite number of linear combinations. However, infinite does not mean any--the linear combinations span a space defined by the original vectors. In this case, we would create a plane as you can start to see above. Formally, this illustrates how vectors $\vec{x}$ and $\vec{1}$ span a vector space.
+
+Recall:
+- our scatter plot
+- $X$ and $\vec{y}$ are defined by the data
+- our loss value is the L2 norm of $y-X \hat{\theta}$, squared. 
+
+By inspection, we see that no line can perfectly fit our points, so we cannot achieve 0 loss. Thus, we know that $\vec{y}$ is not in the plane spanned by $\vec{x}$ and $\vec{1}$.
+
+Visually,
+
+<img src="../../notebooks-images/linear_projection1.png" width="500" />
+
+Since our error is based on distance, we can see that to minimize $ L(\hat{\theta}, \vec{x}, \vec{y}) = \left \Vert  \vec{y} - X \hat{\theta} \right \Vert ^2$, we want $X \hat{\theta}$ to be as close to $\vec{y}$ as possible.
+
+Mathematically, we are looking for the projection of $\vec{y}$ onto the vector space spanned by the columns of $X$ because the projection $\hat{y}$ has the property of being the closest point in the vector space to the vector $y$.
```

Comment:
Hmm.. might be good to explain what the parallelogram is in the diagram (the plane spanned by x and 1)